k256/src/arithmetic/field.rs

```diff
@@ -1,583 +1,289 @@
-    type Output = FieldElement;
+impl Default for FieldElement {
+    fn default() -> Self {
+        Self::zero()
+    }
+}
 
-    fn add(self, other: &FieldElement) -> FieldElement {
-        FieldElement::add(&self, other)
+impl ConditionallySelectable for FieldElement {
+    fn conditional_select(a: &Self, b: &Self, choice: Choice) -> Self {
+        Self(FieldElementImpl::conditional_select(&(a.0), &(b.0), choice))
     }
 }
 
-impl AddAssign<FieldElement> for FieldElement {
-    fn add_assign(&mut self, rhs: FieldElement) {
-        *self = FieldElement::add(self, &rhs);
+impl ConstantTimeEq for FieldElement {
+    fn ct_eq(&self, other: &Self) -> Choice {
+        self.0.ct_eq(&(other.0))
     }
 }
 
-impl Sub<&FieldElement> for &FieldElement {
+impl Add<&FieldElement> for &FieldElement {
     type Output = FieldElement;
 
-    fn sub(self, other: &FieldElement) -> FieldElement {
-        FieldElement::subtract(self, other)
+    fn add(self, other: &FieldElement) -> FieldElement {
+        FieldElement(self.0.add(&(other.0)))
     }
 }
 
-impl Sub<&FieldElement> for FieldElement {
+impl Add<&FieldElement> for FieldElement {
     type Output = FieldElement;
 
-    fn sub(self, other: &FieldElement) -> FieldElement {
-        FieldElement::subtract(&self, other)
+    fn add(self, other: &FieldElement) -> FieldElement {
+        FieldElement(self.0.add(&(other.0)))
     }
 }
 
-impl SubAssign<FieldElement> for FieldElement {
-    fn sub_assign(&mut self, rhs: FieldElement) {
-        *self = FieldElement::subtract(self, &rhs);
+impl AddAssign<FieldElement> for FieldElement {
+    fn add_assign(&mut self, rhs: FieldElement) {
+        *self = *self + &rhs;
     }
 }
 
 impl Mul<&FieldElement> for &FieldElement {
     type Output = FieldElement;
 
     fn mul(self, other: &FieldElement) -> FieldElement {
-        FieldElement::mul(self, other)
+        FieldElement(self.0.mul(&(other.0)))
     }
 }
 
 impl Mul<&FieldElement> for FieldElement {
     type Output = FieldElement;
 
     fn mul(self, other: &FieldElement) -> FieldElement {
-        FieldElement::mul(&self, other)
+        FieldElement(self.0.mul(&(other.0)))
     }
 }
 
 impl MulAssign<FieldElement> for FieldElement {
     fn mul_assign(&mut self, rhs: FieldElement) {
-        *self = FieldElement::mul(self, &rhs);
+        *self = *self * &rhs;
     }
 }
 
-impl Neg for FieldElement {
-    type Output = FieldElement;
-
-    fn neg(self) -> FieldElement {
-        FieldElement::zero() - &self
+#[cfg(test)]
+impl From<&BigUint> for FieldElement {
+    fn from(x: &BigUint) -> Self {
+        debug_assert!(x < &Self::modulus_as_biguint());
+        Self(FieldElementImpl::from(x))
     }
 }
 
-impl<'a> Neg for &'a FieldElement {
-    type Output = FieldElement;
-
-    fn neg(self) -> FieldElement {
-        FieldElement::zero() - self
+#[cfg(test)]
+impl ToBigUint for FieldElement {
+    fn to_biguint(&self) -> Option<BigUint> {
+        self.normalize().0.to_biguint()
     }
 }
```

Comment:
I believe you can move these `#[cfg(test)]`-gated impls into the `tests` module so everything test-related is in one module

Reply:
Would it be useful to expose conversion to/from BigUint in the main library?

Reply:
I don't see a reason to. You're just using it for equivalence testing, right?

Reply:
Got it. Yes, just as a reference in tests instead of fiat-crypto (which is Montgomery-based).

Reply:
Fixed.